### DISCUSSION TITLE
How to use CJS JavaScript file containing `require` in Vue component

### DISCUSSION THREAD

**Discussion by TheAlexLichter** [nuxt/nuxt#17506] [Questions] [answered]
Hey 👋 

In some scenarios, it is not possible to avoid `require` in a JavaScript file (e.g. if it is also used as part of a PostCSS plugin).

**How is it possible to use such CJS files inside Vue components?**

[Example scenario on StackBlitz](https://stackblitz.com/edit/nuxt-starter-fhy5rc?file=config.js,app.vue,nuxt.config.ts)

---

Importing the JavaScript file "as usual" does not work and will yield

> [nuxt] [request error] require is not defined in ES module scope, you can use import instead
  at _evaluate

```vue
<script setup>
import config from './config.js';
</script>

<template>
  <div>
    Can you see this?
    <pre>{{ config }}</pre>
  </div>
</template>
```

Config file with `require`

```js
const colors = require('tailwindcss/colors');

module.exports = {
  themes: [
    {
      name: 'dark',
      theme: {
        colors: {
          body: colors.slate['800'],
        },
      },
    },
  ],
}
```

**Comment by pi0** [comment, marked as answer]
General answer: It is probably best to always ask upstream maintainers for proper ESM support. We have a dedicated [docs page](https://v3.nuxtjs.org/guide/going-further/esm) about this and Nuxt has the best efforts CJS support with Webpack and Nitro (rollup) but I guess your issue originates from Vite which is not compatible with CJS and `require` and adding more CJS support involves lots of hacks considering Vite and whole ecosystem is shifting towards ESM.

For tailwind, have you tried tailwind module with `exposeConfig`? (https://tailwindcss.nuxtjs.org/tailwind/config#referencing-in-the-application). Module reads the config and injects it to a separate chunk to use.

**Reply by TheAlexLichter** [reply]
This seems reasonable! 👍🏻

Thanks for the detailed answer ☺️

Yes, `exposeConfig` will likely be a good workaround for the issue above
